runtime/autoload/man.vim

```diff
@@ -79,43 +79,61 @@ function! man#read_page(ref) abort
   call s:put_page(page)
 endfunction
 
-" Handler for s:system() function.
-function! s:system_handler(jobid, data, event) dict abort
-  if a:event is# 'stdout' || a:event is# 'stderr'
-    let self[a:event] .= join(a:data, "\n")
-  else
-    let self.exit_code = a:data
+" Run man and timeout after 30 seconds.
+"    term          : Term to search.
+"    first vararg  : List of extra arguments to pass to man.
+"    second vararg : Dictionary of environment variables.
+function! s:run_man(term, ...) abort
+  if a:0 > 2
+    throw 'Too many arguments'
   endif
-endfunction
 
-" Run a system command and timeout after 30 seconds.
-function! s:system(cmd, ...) abort
+  let args = get(a:000, 0, [])
+  let env  = get(a:000, 1, {})
+
+  let cmd = ['man'] + args
+  if !empty(a:term)
+    let cmd += ['--', a:term]
```

Comment:
Yea like you mentioned in your commit I'm not sure if this will be portable enough. We'd need to test across all supported platforms before I would be comfortable approving this.